soc/intel/intel_adsp/common/include/intel_adsp_hda.h

```diff
@@ -283,7 +283,7 @@ static inline uint32_t intel_adsp_hda_unused(uint32_t base, uint32_t regblock_si
 	int32_t wp = *DGBWP(base, regblock_size, sid);
 	int32_t size = rp - wp;
 
-	if (size <= 0) {
+	if (size < 0) {
```

Comment:
Or the decision should be:
read RP and WP, if RP==WP, read the DGCS and see if it is empty or full?

Reply:
Your change can save two register read most of the time when the DGCS flags are correct, you only read the RP/WP otherwise.

If we go with that, the commit message should be updated, I don't get how we 'prioritize BNE bit' at all, what we are at the end doing is written in the comment of the diff I posted above.
Also, a comment explaining what is going on would not hurt in the code either.

Reply:
Ack @plbossart and @ujfalusi . I now uploaded V2 to use @ujfalusi 's version. This is called once per timer tick, so I don't think avoiding register reads is critical here and reading DGCS seems like the safer approach.